analyses/malawi/dryspells_trigger/07_ecmwf_monthly_skill_general.md

```diff
@@ -41,164 +37,372 @@ reload(processing)
-        crps_rainy = xs.crps_ensemble(
-            observations_rainy,
-            forecast.where(forecast.time.dt.month.isin(subset_dict['rainy']), drop=True),
-            member_dim='number')
-        append_dict.update({
-                f'crps_rainy': crps_rainy.values,
-                f'std_rainy': observations_rainy.std().values,
-                f'mean_rainy': observations_rainy.mean().values
-            })
-    
-    if "thresh" in subset_dict:
-        for thresh in subset_dict["thresh"]:
-            crps_thresh = xs.crps_ensemble(observations.where(observations<=thresh), forecast.where(observations<=thresh), member_dim='number')
-            append_dict.update({
-                f'crps_{thresh}': crps_thresh.values,
-                f'std_{thresh}': observations.where(observations<=thresh).std().values,
-                f'mean_{thresh}': observations.where(observations<=thresh).mean().values
-            })
-        df_crps= df_crps.append([append_dict], ignore_index=True)
-```
-
-```python
-def plot_skill(df_crps, division_key=None,
-              ylabel="CRPS [mm]"):
-    fig, ax = plt.subplots()
-    df = df_crps.copy()
-    for i, subset in enumerate([k for k in df_crps.keys() if "crps" in k]):
-        y = df[subset]
-        if division_key is not None:
-            dkey = f'{division_key}_{subset.split("_")[-1]}' if subset!="crps" else division_key
-            y /= df[dkey]
-        ax.plot(df['leadtime'], y, ls="-", c=f'C{i}')
-    ax.plot([], [], ls="-", c='k')
-    # Add colours to legend
-    for i, subset in enumerate([k for k in append_dict.keys() if "crps" in k]):
-        ax.plot([], [], c=f'C{i}', label=subset)
-    ax.set_title("ECMWF forecast skill in Malawi:\n 2000-2020 forecast")
-    ax.set_xlabel("Lead time (months)")
-    ax.set_ylabel(ylabel)
-    ax.legend()
-```
-
-```python
-#performance pretty bad.. especially looking at the mean values, it is about 20% off on average for decjanfeb..
-# Plot absolute skill
-plot_skill(df_crps)
-
-# Rainy season performs the worst, but this is likely because 
-# the values during this time period are higher. Try using 
-# reduced skill (dividing by standard devation).
-plot_skill(df_crps, division_key='std', ylabel="RCRPS")
-
-#This is perhpas not exactly what we want because we know this 
-#data comes from the same location and the dataset has the same properties, 
-#but we are splitting it up by mean value. Therefore try normalizing using mean
-plot_skill(df_crps, division_key='mean', ylabel="NCRPS (CRPS / mean)")
+Below we plot the range of observed and forecasted values per month, but across all leadtimes. We can see that they both show the same pattern across the months. However, the observed data shows a broader range of values, and especially has more outliers. The medians are relatively in the same range for observed and forecasted values, though when zooming in you can see that they also differ. 
+
+```python
```

Comment:
It might be useful to split up this plot (and the ones below) by leadtime, just to see if there is any difference. It should be more difficult to forecast 6 months ahead than 1 month ahead, right?!

Reply:
Haha yes it should be. 
What I struggle with is that it becomes so much information, which is hard to compare. That is why I tried to summarize the differences in leadtime by the bias. 
I tried separating these plots by leadtime. You do see some differences especially 1 months vs all other months. But comparison is really hard

Reply:
For sure it's a lot of information. But I think it's important to keep the variables separate if you can, to avoid ambiguous / difficult to interpret results due to competing effects between variables. It might help to focus just on one or two lead times that you are interested in, for example